.github/workflows/cache-cleanup.yml

```diff
@@ -0,0 +1,39 @@
+# This workflow cleans up any cache entries associated with a pull request once it has been closed.
+# This prevents us from having many refs/pull/PR_NUMBER/merge cache entries which will never be used.
+#
+# Note that this will affect both PRs being closed with and without being merged.
+
+name: Cleanup closed PR cache entries
+
+on:
+  pull_request:
+    types:
+      - closed
+
+jobs:
+  cleanup:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
```

Comment:
Do we actually need to check out the code?

Reply:
Not sure, doesn't seem like it. I can try removing and then we can re-add if things blow up.